compiler/vm.nim

```diff
@@ -1028,9 +1027,9 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
         if regs[rc].kind == rkNodeAddr:
           ret = regs[rb].nodeAddr == regs[rc].nodeAddr
         else:
-          ret = ptrEquality(regs[rb].nodeAddr, regs[rc].node)
+          ret = ptrEquality(regs[rb].nodeAddr, regs[rc].regToNode) # todo fixme
```

Comment:
I'm checking it...

Reply:
It seems to be known issue:

```nim
type
  SinglyLinkedList[T] = ref object

proc addMoved[T](a, b: var SinglyLinkedList[T]) =
  if a.addr != b.addr: discard

proc main =
  var a: SinglyLinkedList[int]; new a
  var b: SinglyLinkedList[int]; new b
  a.addMoved b

static: main()
```
gives `field 'node' is not accessible for type 'TFullReg' using 'kind = rkRegisterAddr' [FieldDefect]` with 1.6.6

Reply:
=> https://github.com/nim-lang/Nim/issues/19909

Reply:
fixed on the latest commit